admittance_controller/include/admittance_controller/admittance_rule.hpp

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2021, PickNik, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+/// \authors: Denis Stogl, Andy Zelenak, Paul Gesel
+
+#ifndef ADMITTANCE_CONTROLLER__ADMITTANCE_RULE_HPP_
+#define ADMITTANCE_CONTROLLER__ADMITTANCE_RULE_HPP_
+
+#include <map>
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include "tf2_ros/transform_listener.h"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+#include <tf2_ros/buffer.h>
+#include "tf2_kdl/tf2_kdl/tf2_kdl.hpp"
+#include "control_msgs/msg/admittance_controller_state.hpp"
+#include "controller_interface/controller_interface.hpp"
+#include "control_toolbox/filters.hpp"
+#include "geometry_msgs/msg/wrench_stamped.hpp"
+#include "config/admittance_struct.h"
+
+// kinematics plugins
+#include "kinematics_interface/kinematics_interface_base.hpp"
+#include "pluginlib/class_loader.hpp"
+
+
+namespace {  // Utility namespace
+
+// Numerical accuracy checks. Used as deadbands.
+  const double ROT_AXIS_EPSILON = 1e-6;
+
+
+}  // utility namespace
+
+namespace admittance_controller {
+
+  class AdmittanceRule {
+  public:
+    AdmittanceRule() {
+      parameters_ = &parameters_copy;
```

Comment:
I am going to change the API in the parameter handler to avoid this. The only reason `parameters_copy` is a member is so there is allocated memory to copy parameters into if dynamic parameters is disabled. I am going to add a method in the parameter handler class like this `parameters_ = parameters_handler_.get_latest()` and another method like `parameters_handler_.has_updated()`. This way the logic for will be mush less confusing.